дайБог

vvverman opened the pull request:
сБожьейПомощью

vvverman committed:
test

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,23 @@
+
+        backButton.setOnClickListener {
+            val intent = Intent(this, MainActivity::class.java)
+            startActivity(intent)
```
Для перехода назад стоит использовать не интент, а методы finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран, создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на экран настроек

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+
+
+        <Button
+            android:id="@+id/search"
```
Лучше присваивать View id, характеризующие сущность - buttonSearch, editTextSearch и т.д. При этом удобно идти от общего к частному - так удобнее искать схожие элементы, когда начинаешь печатать, чтобы найти элемент, и выпадают подсказки

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+            android:text="@string/search_button"
+            android:textAllCaps="false"
+            android:textColor="@color/black"
+            app:cornerRadius="16dp"
```
Повторяющиеся аналогичные размеры стоит вынести в ресурсы

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_marginTop="33dp"
```
Рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР. Уменьшил до 32 и добавил в ресурсы

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+            app:cornerRadius="16dp"
+            app:icon="@drawable/search_button"
+            app:iconGravity="textStart"
+            app:iconPadding="10dp"
```
В ресурсы

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР.

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,115 @@
+        android:paddingVertical="21dp">
+
+        <TextView
+            android:id="@+id/textView"
```
Не очень удачный нейминг для id - в этом файле 4 TextView

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР. Переименовано в shareButton

vvverman replied:
готово

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,115 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:paddingVertical="21dp">
```
В ресурсы

vvverman replied:
Помечаю для себя как исправленную, залью в ближайший ПР. Уменьшил до 20 и увёл в ресурсы

vvverman replied:
готово

faritowich commented on app/src/main/res/drawable/arrow_back_button.xml:
```diff
@@ -0,0 +1,9 @@
+    android:viewportHeight="16">
+  <path
+      android:pathData="M16,7H3.83L9.42,1.41L8,0L0,8L8,16L9.41,14.59L3.83,9H16V7Z"
+      android:fillColor="#1A1B22"/>
```
Цвет встречается неоднократно, его также можно вынести в ресурсы

vvverman replied:
Не нашёл, где он встречается ещё, но увёл в ресурсы

vvverman replied:
готово

vvverman committed:
первые правки готовы

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+            app:cornerRadius="16dp"
+            app:icon="@drawable/settings_button"
+            app:iconGravity="textStart"
+            app:iconPadding="10dp"
```
Повторяющиеся аналогичные размеры, в ресурсы

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,74 @@
+            android:text="@string/setting_button"
+            android:textAllCaps="false"
+            android:textColor="@color/black"
+            app:cornerRadius="16dp"
```
Повторяющиеся аналогичные размеры, в ресурсы

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,115 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:paddingVertical="21dp">
```
Повторяющиеся аналогичные размеры, в ресурсы

faritowich commented on app/src/main/res/values/dimensions.xml:
```diff
@@ -0,0 +1,7 @@
+    <dimen name="cornerRadius">16dp</dimen>
+    <dimen name="layout_marginTop">32dp</dimen>
+    <dimen name="iconPadding">10dp</dimen>
+    <dimen name="paddingVertical">20dp</dimen>
```
Эти размеры вынесены в ресурсы, но в XML-файлах по факту используются не везде, где надо.

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 1

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 2

Оживил кнопку Write to Support

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 3

Оживил кнопки Share и Terms Of Use

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 4

Сделал строку поиска и кнопку стирания запроса

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 4

Сделал строку поиска и кнопку стирания запроса

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Домашнее задание часть 5

Теперь данные в активити поиска не будут пропадать, если пользователь уйдёт с активити

nadzyathedude commented on app/src/main/res/drawable/search_bg.xml:
```diff
@@ -0,0 +1,16 @@
+
+</shape>
+
+<!--<vector xmlns:android="http://schemas.android.com/apk/res/android"-->
+<!--    android:width="328dp"-->
+<!--    android:height="36dp"-->
+<!--    android:viewportWidth="328"-->
+<!--    android:viewportHeight="36">-->
+<!--  <path-->
+<!--      style="@style/SearchLineBG"-->
+<!--      android:pathData="M8,0L320,0A8,8 0,0 1,328 8L328,28A8,8 0,0 1,320 36L8,36A8,8 0,0 1,0 28L0,8A8,8 0,0 1,8 0z" />-->
+<!--</vector>-->
```
стоить удалить закомментированный код если он не нужен

nadzyathedude commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,109 @@
+            android:visibility="gone"/>
+    </FrameLayout>
+
+<!--    <LinearLayout-->
+<!--        android:id="@+id/container"-->
+<!--        android:layout_width="match_parent"-->
+<!--        android:layout_height="wrap_content"-->
+<!--        android:layout_marginTop="16dp"-->
+<!--        android:orientation="horizontal"-->
+<!--        tools:context=".MainActivity">-->
+
+<!--        <EditText-->
+<!--            android:id="@+id/inputEditText"-->
+<!--            android:paddingLeft="40dp"-->
+<!--            android:layout_width="0dp"-->
+<!--            android:layout_height="wrap_content"-->
+<!--            android:layout_weight="1"-->
+<!--            android:hint="@string/input_hint"-->
+<!--            android:inputType="number"-->
+<!--            android:maxLength="5"-->
+<!--            android:maxLines="1"-->
+<!--            android:textColor="@color/black"-->
+<!--            android:textColorHint="#FFAEAFB4"-->
+<!--            android:background="@drawable/search_bg" />-->
+
+<!--        <ImageView-->
+<!--            android:id="@+id/clearIcon"-->
+<!--            android:layout_width="wrap_content"-->
+<!--            android:layout_height="wrap_content"-->
+<!--            android:layout_gravity="center"-->
+<!--            android:contentDescription="This is button of clear string"-->
+<!--            android:src="@drawable/clear_button"-->
+<!--            android:visibility="gone"-->
+<!--            />-->
```
тут тоже закомментированный код, его тоже стоит удалить. в случае, если тебе понадобиться вернуться к этому коду, можно откатиться с помощью гита. это одна из его главных целей

nadzyathedude commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,49 @@
+    </style>
+
+</resources>
+
+
+
+
+
+
+
+
```
пустые строки стоит удалить

vvverman committed:
Спринт 9/18 → Тема 1/5: Стили и темы → Урок 3/3

Правки

исправлено следующее

1) Цвета, отступы и шрифты соответствуют дизайну
На экране настроек switch, переключающий темную тему выходит на гайдланы.
paddingTop у FrameLayout с поисковиком не соответсвует дизайну.
2)Для «Главного экрана» стиль кнопки наследуется от Widget.MaterialComponents.Button.UnelevatedButton.
Сейчас это просто Button.
3) При нажатии на поле ввода появляется клавиатура.

4) Нажатие на кнопку «Очистить» удаляет текст в поле ввода поискового запроса и скрывает клавиатуру.

5) У кнопки «Очистить» есть OnClickListener, в котором очищается поисковый запрос и прячется клавиатура.

nadzyathedude commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -38,7 +37,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:paddingTop="16dp">
+        android:paddingTop="29dp">
```
[ссылка на дизайн](https://monosnap.com/file/qfVcZBnlmKqlt3iXfJ6WmZqWDi6xju)
